snarkos/ledger/mod.rs

```diff
@@ -287,19 +287,21 @@ impl<N: Network> Ledger<N> {
                         let block_url = format!("{TARGET_URL}{height}.block");
 
                         // Fetch the bytes from the given url
-                        reqwest::get(block_url).await
+                        let block_bytes = reqwest::get(block_url).await?.bytes().await?;
+
+                        // Parse the block.
+                        let block = task::spawn_blocking(move || Block::from_bytes_le(&block_bytes)).await.unwrap()?;
+
+                        std::future::ready(Ok(block)).await
                     })
                 })
                 .buffered(CONCURRENT_REQUESTS)
-                .for_each(|response| async {
+                .for_each(|block| async {
+                    let block = block.unwrap();
```

Comment:
note: this guaranteed to not panic, as `Ok(block)` is passed from the previous step